### PR TITLE
Explorer validator sorting

### PR DESCRIPTION
**Pull request by Jibz1**
https://user-images.githubusercontent.com/8676844/213721479-facc840a-6829-47b8-a34b-d057a0d3ec58.mov

### CODE REVIEW

**Review by mystie711 — APPROVED** [review]
@Jibz1 following up on our conversation yesterday,
- default sort by stake
- remove unsorted state

File: apps/explorer/src/ui/TableCard.tsx [hunk 1]
```diff
@@ -4,10 +4,15 @@
 import {
     flexRender,
     getCoreRowModel,
+    getSortedRowModel,
     useReactTable,
+    //  type ColumnDef,
```
Comment (mamos-mysten) [inline apps/explorer/src/ui/TableCard.tsx:9]: can we make use of this type?

File: apps/explorer/src/ui/TableCard.tsx [hunk 2]
```diff
@@ -46,29 +52,53 @@ type DataType = {
...
+export function TableCard({
+    refetching,
+    data,
+    columns,
+    sortTable,
+    defaultSorting,
+}: TableCardProps) {
     // Use Columns to create a table
-    const processedcol = useMemo(() => columnsContent(columns), [columns]);
+    const processedcol = useMemo(
```
Comment (mamos-mysten) [inline apps/explorer/src/ui/TableCard.tsx:82]: we can improve typings here e.g. `useMemo<ColumnDef<TableData>[]> where `TableData` is a type representing the data in the table

**Review by mamos-mysten — CHANGES_REQUESTED** [review]

File: apps/explorer/src/ui/TableCard.tsx [hunk 3]
```diff
@@ -80,40 +110,79 @@ export function TableCard({ refetching, data, columns }: TableCardProps) {
...
-                                        cell.getContext()
-                                    )}
-                                </td>
-                            ))}
+                            {row
+                                .getVisibleCells()
+                                .map(({ column, id, getContext }) => (
+                                    <td
+                                        key={id}
+                                        className="h-7.5 px-1 text-body text-gray-75 group-hover:bg-gray-40 group-hover:text-gray-90 group-hover:first:rounded-l group-hover:last:rounded-r"
```
[the remark this answers is not in the document]
Comment (Jibz1) [reply apps/explorer/src/ui/TableCard.tsx:178]: I think some of the transaction fields use the default style. Like the `gas` fields. The current implementation of the table data needs to be corrected. We must pass the data to the table data field and components in the column. I will create separate takes for that. Once we have that, we can remove the text style.

[the remark this answers is not in the document]
Comment (Jordan-Mysten) [reply apps/explorer/src/ui/TableCard.tsx:178]: I'll probably pull together a rework for the table in another PR, so don't stress about that.

File: apps/explorer/src/ui/TableCard.tsx [hunk 4]
```diff
@@ -80,40 +110,79 @@ export function TableCard({ refetching, data, columns }: TableCardProps) {
...
+                                        >
+                                            {isPlaceholder
+                                                ? null
+                                                : flexRender(
+                                                      column.columnDef.header,
+                                                      getContext()
+                                                  )}
+                                            {{
+                                                asc: (
+                                                    <ArrowRight
```
[the remark this answers is not in the document]
Comment (Jibz1) [reply apps/explorer/src/ui/TableCard.tsx]: done

File: apps/explorer/src/ui/TableCard.tsx (hunk 1 above)
Comment (Jibz1) [reply apps/explorer/src/ui/TableCard.tsx]: done

File: apps/explorer/src/pages/validators/Validators.tsx [hunk 5]
```diff
@@ -17,16 +17,127 @@ import { Banner } from '~/ui/Banner';
...
+                                    fallback={name}
+                                    circle
+                                />
+                                <Text
+                                    variant="bodySmall/medium"
+                                    color="steel-darker"
+                                >
+                                    {name}
+                                </Text>
+                            </div>{' '}
```
Comment (Jordan-Mysten) [inline apps/explorer/src/pages/validators/Validators.tsx:88]: Extra space

File: apps/explorer/src/pages/validators/Validators.tsx [hunk 6]
```diff
@@ -17,16 +17,127 @@ import { Banner } from '~/ui/Banner';
...
+                lastEpoch:
+                    validator.fields.delegation_staking_pool.fields
+                        .rewards_pool,
+            };
+        }),
+        columns: [
+            {
+                header: '#',
+                accessorKey: 'number',
+                cell: (props: any) => (
```
Comment (Jordan-Mysten) [inline apps/explorer/src/pages/validators/Validators.tsx:56]: Don't love the `any`'s here. We can follow-up later with an improved Table component, just want to call it out

File: apps/explorer/src/pages/validators/Validators.tsx [hunk 7]
```diff
@@ -47,91 +169,36 @@ function ValidatorPageResult() {
         const validatorsApy = validators.map((av) =>
             calculateAPY(av, +validatorsData.epoch)
         );
-        return (
-            validatorsApy.reduce((acc, cur) => acc + cur, 0) /
-            validatorsApy.length
+        return parseFloat(
```
Comment (Jordan-Mysten) [inline apps/explorer/src/pages/validators/Validators.tsx:172]: Probably generalize into function.

File: apps/explorer/src/ui/TableCard.tsx [hunk 8]
```diff
@@ -80,40 +66,79 @@ export function TableCard({ refetching, data, columns }: TableCardProps) {
...
+                                }) => (
+                                    <th
+                                        key={id}
+                                        colSpan={colSpan}
+                                        scope="col"
+                                        className="h-7.5 px-1 text-left text-subtitle font-semibold uppercase text-steel-dark"
+                                        onClick={
+                                            column.columnDef.enableSorting
+                                                ? column.getToggleSortingHandler()
+                                                : void 0
```
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/TableCard.tsx:87]: Just use `undefined`?

File: apps/explorer/src/ui/TableCard.tsx [hunk 9]
```diff
@@ -80,40 +66,79 @@ export function TableCard({ refetching, data, columns }: TableCardProps) {
...
+                                                        className="-rotate-90 text-steel-darker"
+                                                    />
+                                                ),
+                                                desc: (
+                                                    <ArrowRight12
+                                                        fill="currentColor"
+                                                        className="rotate-90 text-steel-darker"
+                                                    />
+                                                ),
+                                            }[column.getIsSorted() as string] ??
```
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/TableCard.tsx:117]: Just make a component <SortIcon /> that can take in a prop that's `asc` or `desc` and return the correct icon based on that. Will be easier to follow.

**Review by Jordan-Mysten — APPROVED** [review]
Approving to unblock but some nits

**Pull request merged**